crates/diesel_models/src/query/payment_attempt.rs

```diff
@@ -412,6 +412,64 @@ impl PaymentAttempt {
         ))
     }
 
+    #[cfg(feature = "v2")]
+    #[allow(clippy::too_many_arguments)]
+    pub async fn get_total_count_of_attempts(
+        conn: &PgPooledConn,
+        merchant_id: &common_utils::id_type::MerchantId,
+        active_attempt_ids: &[String],
+        connector: Option<String>,
+        payment_method_type: Option<enums::PaymentMethod>,
+        payment_method_subtype: Option<enums::PaymentMethodType>,
+        authentication_type: Option<enums::AuthenticationType>,
+        merchant_connector_id: Option<common_utils::id_type::MerchantConnectorAccountId>,
+        card_network: Option<enums::CardNetwork>,
+    ) -> StorageResult<i64> {
+        let mut filter = <Self as HasTable>::table()
+            .count()
+            .filter(dsl::merchant_id.eq(merchant_id.to_owned()))
+            .filter(dsl::id.eq_any(active_attempt_ids.to_owned()))
+            .into_boxed();
+
+        if let Some(connector) = connector {
+            filter = filter.filter(dsl::connector.eq(connector));
+        }
+
+        if let Some(payment_method_type) = payment_method_type {
+            filter = filter.filter(dsl::payment_method_type_v2.eq(payment_method_type));
+        }
+        if let Some(payment_method_subtype) = payment_method_subtype {
+            filter = filter.filter(dsl::payment_method_subtype.eq(payment_method_subtype));
+        }
+        if let Some(authentication_type) = authentication_type {
+            filter = filter.filter(dsl::authentication_type.eq(authentication_type));
+        }
+        if let Some(merchant_connector_id) = merchant_connector_id {
+            filter = filter.filter(dsl::merchant_connector_id.eq(merchant_connector_id))
+        }
+        if let Some(card_network) = card_network {
+            filter = filter.filter(dsl::card_network.eq(card_network))
+        }
+
+        router_env::logger::debug!(query = %debug_query::<Pg, _>(&filter).to_string());
+
+        // TODO: Remove these logs after debugging the issue for delay in count query
```

Comment:
What was this delay in count query?

Reply:
Taken from v1, for reference: #6331 

Should this be removed?